### PR TITLE
Created a settings folder with environment-specific settings files

### LINKED ISSUE #25: Separate settings into multiple files as per build environment
All the settings for the project are in a single file, which is not suitable if different environments for development, testing and production are to be used.

### PR DESCRIPTION
**Pull request by nitinpaultifr**
Using a single settings file for all environments (development, testing, production) is inefficient since many variables and parameters vary according to the current environment. For example, you'd want to set the 'DEBUG' option to 'True' during local development, but it must set to to 'False' in a production environment. This hierarchy lets you do just that. Additional instructions to configure your vitualenv to automatically source respective settings will be provided in the wiki.

Resolves #25 

Signed-off-by: Nitin Paul npaul2811@gmail.com

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ cache:
 env:
   matrix:
     # django
-    - DJANGO_VERSION=1.8 DJANGO_SETTINGS_MODULE="contributr.settings"
+    - DJANGO_VERSION=1.8 DJANGO_SETTINGS_MODULE="contributr.settings.testing"
```
Comment (nitinpaultifr) [inline .travis.yml:20]: Can someone verify if this is the right way to point it to the settings folder? the settings folder is a module, I don't know why it won't see it.

**Comment by tobias47n9e** [comment]
@npaul2811 Could you still try to merge or squash all of those commits into one? It will keep the history a little cleaner and this PR does have a lot of commits: https://github.com/Djenesis/contributr/wiki/Pull-Request-Guidelines

File: .travis.yml (hunk 1 above)
Comment (coolharsh55) [inline .travis.yml:20]: line 20 says that the settings are in a file called `testing.py` within the folder `contribute/settings`. The `settings` folder should have a `__init__.py` file for it to be used as a module.

**Comment by coolharsh55** [comment]
Closing this PR due to:
1. large number of commits
2. unclear commit messages
3. changes introduced are not clear

File: .travis.yml (hunk 1 above)
Comment (nitinpaultifr) [inline .travis.yml:20]: It has one @coolharsh55

**Pull request closed without merging**